### PR TITLE
[WEB-1079] Site & Language Selectors UX overhaul

### PR DESCRIPTION
**Pull request by bgdeutsch**
### What does this PR do?
Design/UX overhaul for the `Site` and `Language` dropdowns

### Motivation
https://datadoghq.atlassian.net/browse/WEB-1079

### Preview
https://docs-staging.datadoghq.com/brian.deutsch/site-lang-toggle2/
https://docs-staging.datadoghq.com/brian.deutsch/site-lang-toggle2/api/

### Additional Notes
This PR introduces one new frontmatter param and a change to an existing param:
- `disable_sidebar`: This removes the sidebar (containing `Site` + `Language` dropdowns, `Edit` button, and TOC) entirely.  Main content will expand to occupy the 2 columns reserved for the sidebar.
- `disable_toc`: This removes _only_ the TOC + `Edit` button, leaving the `Site` + `Language` dropdowns.

Related synthetics (Note these are not scheduled and have only been run ad-hoc.  If there are no results try broadening the search time range):
- Region selector (US/GOV): https://dd-corpsite.datadoghq.com/synthetics/details/yq3-qs9-b8k
- Region selector(EU): https://dd-corpsite.datadoghq.com/synthetics/details/3cj-f8n-a3y
- Language selector: https://dd-corpsite.datadoghq.com/synthetics/details/4rx-amr-pnv

There are also Jest unit tests.   Pull down this branch and from the `documentation` root: `jest src/scripts/tests/region-redirects.test.js` to run.   There is some duplication amongst the synthetics and jest tests, which will be addressed at a later time.

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by l0k0ms** [comment]
It seems that there are some ToC overflow issues w/ very long ToC: https://docs-staging.datadoghq.com/brian.deutsch/site-lang-toggle2/logs/processing/attributes_naming_convention/

File: src/scripts/region-redirects.js [hunk 1]
```diff
@@ -3,14 +3,28 @@ import config from '../../regions.config';
...
+                const region = option.dataset.value;
+                regionOnChangeHandler(region)
+            })
+        })
     }
 });
 
-function isReferrerEU(){
+function replaceButtonInnerText(value) {
+    const selectedRegion = value === 'gov' ? 'US1-FED' : value.toUpperCase();
```
Comment (nsollecito) [inline src/scripts/region-redirects.js:21]: can we set `selectedRegion` here from the config value dd_datacenter?

Comment (bgdeutsch) [reply src/scripts/region-redirects.js]: @nsollecito this is much better.  thanks!

**Comment by bgdeutsch** [comment]
> It seems that there are some ToC overflow issues w/ very long ToC: https://docs-staging.datadoghq.com/brian.deutsch/site-lang-toggle2/logs/processing/attributes_naming_convention/#default-standard-attribute-list

Thanks @l0k0ms!   Made a fix here so long TOC won't overflow into the footer.

**Review by cswatt — APPROVED** [review]
this looks great!

**Pull request merged**